### PR TITLE
Generate table row ids

### PR DESCRIPTION
**Pull request by AlanCornthwaiteKatalyst**

### CODE REVIEW

File: app/components/katalyst/table_component.rb [hunk 1]
```diff
@@ -85,6 +89,10 @@ def sorting
       collection.sorting if collection.respond_to?(:sorting)
     end
 
+    def generate_ids?
+      @generate_ids
```
Comment (hasarindaKI) [inline app/components/katalyst/table_component.rb:93]: is it `!!@generate_ids`?

Comment (hasarindaKI) [reply app/components/katalyst/table_component.rb:93]: or `@generate_ids.present?`? 

https://stackoverflow.com/questions/28498342/converting-anything-to-a-boolean-in-ruby

**Review by hasarindaKI — CHANGES_REQUESTED** [review]

File: app/components/katalyst/table_component.rb (hunk 1 above)
Comment (AlanCornthwaiteKatalyst) [reply app/components/katalyst/table_component.rb:93]: I've gone with `@generate_ids.present?` to match the others in the class

File: app/components/katalyst/table_component.rb [hunk 2]
```diff
@@ -37,6 +38,7 @@ def initialize(collection:,
                    sorting: nil,
                    header: true,
                    caption: true,
+                   generate_ids: true,
```
Comment (sfnelson) [inline app/components/katalyst/table_component.rb:41]: I don't want this to be on by default – my philosophy with tables is that they should generate absolutely minimal syntax. I think that adding IDs is a niche use-case – as we saw it's not even required for morphing.

**Pull request merged**